src/distributions/uniform.rs

```diff
@@ -826,12 +826,109 @@ pub struct UniformFloat<X> {
     scale: X,
 }
 
+trait Summable<T> {
+    fn compensated_sum(&self) -> T;
+}
+
+trait ScaleComputable<T> {
+    fn compute_scale(low: T, high: T) -> T;
+}
```

Comment:
Could you put this in `distributions/utils.rs`? `uniform.rs` is already big enough!

Reply:
Quick question (for anyone familiar with this neighborhood of the code): Would there be a problem replacing the expressions `::core::$u_scalar::MAX >> $bits_to_discard).into_float_with_exponent(0) - 1.0` with `1.0 as $f_scalar - $f_scalar::EPSILON`?  It gives the same result, and doesn't require `$bits_to_discard` or the `IntoFloat` trait.

Reply:
No, please go ahead.